sensors/nxt_i2c_sensor_defs.c

```diff
@@ -446,6 +476,61 @@ const struct nxt_i2c_sensor_info nxt_i2c_sensor_defs[] = {
 			},
 		},
 	},
+	[LEGO_NXT_TEMPERATURE_SENSOR] = {
+		/**
+		 * @vendor_name: LEGO
+		 * @vendor_part_number: 9749
+		 * @vendor_part_name: MINDSTORMS Temperature Sensor
+		 * @vendor_website: https://education.lego.com/en-us/products/mindstorms-temperature-sensor/9749
+		 * @default_address: 0x4C
+		 */
+		.name		= LEGO_NXT_TEMPERATURE_SENSOR_NAME,
+		.vendor_id	= "LEGO",
+		.product_id	= "", /* LMS2012 fakes this with "Temp." */
+		.num_modes	= 2,
+		.mode_info	= (const struct lego_sensor_mode_info[]) {
+			[0] = {
+				/**
+				 * @description: Continuous measurement
+				 * @value0: Temperature (-55.0 to 128.0)
+				 * @units_description: °C
+				 */
+				.name	= "NXT-TEMP-C",
+				.units	= "C",
+				.raw_min = -14080,
+				.raw_max = 32767,
+				.si_min = -55,
+				.si_max = 128,
+				.decimals = 1,
+				.data_type = LEGO_SENSOR_DATA_S16_BE,
+				.scale	= lego_nxt_temp_celsius_scale,
+			},
+			[1] = {
+				/**
+				 * @description: Continuous measurement
+				 * @value0: Temperature (-67.0 to 262.4)
+				 * @units_description: °F
+				 */
+				.name	= "NXT-TEMP-F",
+				.units	= "F",
+				.raw_min = -14080,
+				.raw_max = 32767,
+				.si_min = -67,
+				.si_max = 262,
+				.decimals = 1,
+				.data_type = LEGO_SENSOR_DATA_S16_BE,
+				.scale	= lego_nxt_temp_fahrenheit_scale,
+			}
+		},
+		.i2c_mode_info	= (const struct nxt_i2c_sensor_mode_info[]) {
+			[0] = {
+				.read_data_reg	= 0x00,
```

Comment:
This may seem redundant since we are already setting the mode (write 0x60 to register 0x01) when initializing the sensor, but they should be added here too. These definitions are actually used for other devices as well that don't use the kernel i2c subsystem. So having `set_mode_reg` and `set_mode_data` here will allow the driver to be used in those cases.